src/Microsoft.ML.TorchSharp/Utils/FileUtils.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+
+namespace Microsoft.ML.TorchSharp.Utils
+{
+    internal static class FileUtils
+    {
+        private const int WriteBatchSize = 1024 * 1024;     // this size seems good on performance
+        private static readonly Type[] _validTypes = new Type[]
+        {
+            typeof(short),
+            typeof(ushort),
+            typeof(int),
+            typeof(uint),
+            typeof(long),
+            typeof(ulong),
+            typeof(float),
+            typeof(double),
+        };
+
+        public static string LoadFromFileOrDownloadFromWeb(string path, string fileName, Uri url)
+        {
+            var contents = "";
+            var filePath = Path.Combine(path, fileName);
+            if (!File.Exists(filePath))
+            {
+                try
+                {
+                    using var webClient = new WebClient();
```

Comment:
>WebClient [](http://example.com/codeflow?start=46&length=9)

Our docs is not recommending using this class anymore [https://docs.microsoft.com/en-us/dotnet/api/system.net.webclient?view=net-6.0#remarks](https://docs.microsoft.com/en-us/dotnet/api/system.net.webclient?view=net-6.0#remarks).

Should we consider using HttpClient  instead?

Reply:
I switched to use HttpClient.

Reply:
So switching actually causes some problems. It doesn't download a string identically to how the WebClient does. This code is only used by the BpeTokenizer and so it will go away when we swtich to your new implementation. As such I'm going to leave it as a WebClient for now knowing it is only temporary.